Fails to build with Clang

From [error log](https://github.com/yuzu-emu/yuzu/files/5491366/yuzu-qt5-s20201104.log):
```c++
In file included from src/common/fiber.cpp:6:
src/./common/fiber.h:44:5: error: explicitly defaulted move constructor is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
    Fiber(Fiber&&) = default;
    ^
src/./common/fiber.h:81:14: note: move constructor of 'Fiber' is implicitly deleted because field 'guard' has a deleted move constructor
    SpinLock guard{};
             ^
src/./common/spin_lock.h:23:22: note: copy constructor of 'SpinLock' is implicitly deleted because field 'lck' has a deleted copy constructor
    std::atomic_flag lck = ATOMIC_FLAG_INIT;
                     ^
/usr/include/c++/v1/atomic:2531:5: note: 'atomic_flag' has been explicitly marked deleted here
    atomic_flag(const atomic_flag&) = delete;
    ^
In file included from src/core/cpu_manager.cpp:17:
In file included from src/./video_core/gpu.h:17:
src/./video_core/dma_pusher.h:77:32: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
static constexpr CommandHeader BuildCommandHeader(BufferMethods method, u32 arg_count,
                               ^
src/./video_core/dma_pusher.h:80:19: note: member call on member 'method' of union with active member 'argument' is not allowed in a constant expression
    result.method.Assign(static_cast<u32>(method));
                  ^
In file included from src/core/cpu_manager.cpp:17:
src/./video_core/gpu.h:276:40: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
        static constexpr CommandHeader Build(FenceOperation op, u32 syncpoint_id) {
                                       ^
src/./video_core/gpu.h:278:23: note: member call on member 'op' of union with active member 'raw' is not allowed in a constant expression
            result.op.Assign(op);
                      ^
```

--- a/src/common/fiber.h
+++ b/src/common/fiber.h
@@ -41,8 +41,8 @@ public:
     Fiber(const Fiber&) = delete;
     Fiber& operator=(const Fiber&) = delete;
 
-    Fiber(Fiber&&) = default;
-    Fiber& operator=(Fiber&&) = default;
+    Fiber(Fiber&&) = delete;
+    Fiber& operator=(Fiber&&) = delete;
 
     /// Yields control from Fiber 'from' to Fiber 'to'
     /// Fiber 'from' must be the currently running fiber.

--- a/src/common/spin_lock.h
+++ b/src/common/spin_lock.h
@@ -15,6 +15,14 @@ namespace Common {
  */
 class SpinLock {
 public:
+    SpinLock() = default;
+
+    SpinLock(const SpinLock&) = delete;
+    SpinLock& operator=(const SpinLock&) = delete;
+
+    SpinLock(SpinLock&&) = delete;
+    SpinLock& operator=(SpinLock&&) = delete;
+
     void lock();
     void unlock();
     [[nodiscard]] bool try_lock();

--- a/src/video_core/dma_pusher.h
+++ b/src/video_core/dma_pusher.h
@@ -18,6 +18,8 @@ class System;
 
 namespace Tegra {
 
+class GPU;
+
 enum class SubmissionMode : u32 {
     IncreasingOld = 0,
     Increasing = 1,
@@ -74,16 +76,13 @@ union CommandHeader {
 static_assert(std::is_standard_layout_v<CommandHeader>, "CommandHeader is not standard layout");
 static_assert(sizeof(CommandHeader) == sizeof(u32), "CommandHeader has incorrect size!");
 
-static constexpr CommandHeader BuildCommandHeader(BufferMethods method, u32 arg_count,
-                                                  SubmissionMode mode) {
+inline CommandHeader BuildCommandHeader(BufferMethods method, u32 arg_count, SubmissionMode mode) {
     CommandHeader result{};
     result.method.Assign(static_cast<u32>(method));
     result.arg_count.Assign(arg_count);
     result.mode.Assign(mode);
     return result;
 }
-
-class GPU;
 
 struct CommandList final {
     CommandList() = default;

--- a/src/video_core/gpu.h
+++ b/src/video_core/gpu.h
@@ -273,7 +273,7 @@ public:
         BitField<0, 1, FenceOperation> op;
         BitField<8, 24, u32> syncpoint_id;
 
-        static constexpr CommandHeader Build(FenceOperation op, u32 syncpoint_id) {
+        static CommandHeader Build(FenceOperation op, u32 syncpoint_id) {
             FenceAction result{};
             result.op.Assign(op);
             result.syncpoint_id.Assign(syncpoint_id);

